source/kernel/src/trace.rs

```diff
@@ -195,3 +310,25 @@ impl Collect for SerialCollector {
         false
     }
 }
+
+const fn level_to_u8(level: LevelFilter) -> u8 {
+    match level {
+        LevelFilter::TRACE => 0,
+        LevelFilter::DEBUG => 1,
+        LevelFilter::INFO => 2,
+        LevelFilter::WARN => 3,
+        LevelFilter::ERROR => 4,
+        LevelFilter::OFF => 5,
+    }
+}
+
+const fn u8_to_level(level: u8) -> LevelFilter {
```

Comment:
It *might* be good to have an invalid level be an error (and ignore the "change log level command" instead), otherwise if we receive a garbage message that HAPPENS to look like a command, we have a 251/256 chance of just shutting off all logging. Tho I guess we don't really need to log at all to a connection that is sending us garbage. Hm.

Reply:
these functions are used internally for converting the `AtomicU8` to/from a `LevelFilter`, not for parsing the log level sent to us from the host. that gets deserialized using `postcard`, so if we get an invalid value, we will error.